new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleTraefikValidation.java

```diff
@@ -0,0 +1,209 @@
+import static oracle.weblogic.kubernetes.assertions.TestAssertions.traefikIsRunning;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.awaitility.Awaitility.with;
+
+// this is an example of a test suite (class) where the tests need to be run in a certain
+// order. this is controlled with the TestMethodOrder annotation
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@DisplayName("Simple validation of Traefik functions")
+@ExtendWith(Timing.class)
+class ItSimpleTraefikValidation implements LoggedTest {
+
+  private HelmParams tfHelmParams = null;
+  private HelmParams ingressParam = null;
+  private String tfNamespace = null;
+  private String domainNamespace1 = null;
+
+  @Test
+  @Order(1)
+  @DisplayName("Install the Traefik")
+  // tags are used to filter which tests to run, we can define whatever tags we need,
+  // like these two:
+  @Slow
+  @MustNotRunInParallel
+  public void testInstallingTraefik() {
+
+    // get a new unique traefik Namespace
+    tfNamespace = createNamespace();
+    logger.info(String.format("Created a new namespace called %s", tfNamespace));
+
+    String projectRoot = System.getProperty("user.dir");
+
+    // helm install parameters
+    tfHelmParams = new HelmParams()
+        .releaseName("traefik-operator")
+        .namespace(tfNamespace)
+        .repoUrl("https://kubernetes-charts.storage.googleapis.com/")
+        .repoName("stable")
+        .chartName("traefik")
+        .chartValuesFile(projectRoot + "/../kubernetes/samples/charts/traefik/values.yaml");
+
+    TraefikParams tfParams = new TraefikParams()
+        .helmParams(tfHelmParams)
+        .nameSpaces("{" + tfNamespace + "}");
+
+    // install Traefik
+    assertThat(installTraefik(tfParams))
+        .as("Test installTraefik returns true")
+        .withFailMessage("installTraefik() did not return true")
+        .isTrue();
+    logger.info(String.format("Traefik is installed in namespace %s", tfNamespace));
+
+    assertThat(helmList(tfHelmParams))
+        .as("Test helmList returns true")
+        .withFailMessage("helmList() did not return true")
+        .isTrue();
+
+    // waiting for an async operation to complete.
+    // we first wait 30 seconds, then we check if the traefik pod is running.
+    with().pollDelay(30, SECONDS)
+        // we check again every 10 seconds.
+        .and().with().pollInterval(10, SECONDS)
+        // this listener lets us report some status with each poll
+        .conditionEvaluationListener(
+            condition -> logger.info(
+                "Waiting for traefik to be running (elapsed time {0}ms, remaining time {1}ms)",
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        // and here we can set the maximum time we are prepared to wait
+        .await().atMost(5, MINUTES)
+        .until(traefikIsRunning(tfNamespace));
```

Comment:
Are we waiting for the pod to be running or for its condition to be ready?

Reply:
added a method to check the condition is ready